lib/Sources/BluetoothAction/GetDevices.swift

```diff
@@ -34,8 +34,16 @@ struct GetDevices: BluetoothAction {
     }
 
     func execute(state: BluetoothState, client: BluetoothClient) async throws -> GetDevicesResponse {
-        let uuids = await state.getKnownPeripheralIdentifiers()
-        let peripherals = await client.getPeripherals(withIdentifiers: uuids)
+        let knownUuids = await state.getKnownPeripheralIdentifiers()
+        let peripherals = await client.getPeripherals(withIdentifiers: Array(knownUuids))
+
+        // As a side effect, execute should remove any known peripheral identifiers that could not
+        // be returned by client from persistence
+        let uuidsToForget = knownUuids.subtracting(peripherals.map { $0.id })
+        for uuidToForget in uuidsToForget {
+            await state.forgetPeripheral(identifier: uuidToForget)
+        }
```

Comment:
Perhaps a nit/over-optimization, but think this logic is best pushed down one layer by adding a plural method to `BluetoothState` e.g.:
```suggestion
        await state.forgetPeripherals(identifiers: uuidsToForget)
```